src/utils/components/CPUMemoryModal/CpuMemoryModal.tsx

```diff
@@ -0,0 +1,200 @@
+import * as React from 'react';
+import produce from 'immer';
+
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { k8sUpdate } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  Dropdown,
+  DropdownItem,
+  DropdownToggle,
+  Modal,
+  ModalVariant,
+  NumberInput,
+  Title,
+  TitleSizes,
+} from '@patternfly/react-core';
+
+import useTemplateDefaultCpuMemory from './hooks/useTemplateDefaultCpuMemory';
+import { getCPUCount, getMemorySize, memorySizesTypes } from './utils/CpuMemoryUtils';
+
+import './cpu-memory-modal.scss';
+
+type CPUMemoryModalProps = {
+  vm: V1VirtualMachine;
+  isOpen: boolean;
+  onClose: () => void;
+};
+
+const CPUMemoryModal: React.FC<CPUMemoryModalProps> = ({ vm, isOpen, onClose }) => {
+  const { t } = useKubevirtTranslation();
+  const {
+    data: templateDefaultsData,
+    loaded: defaultsLoaded,
+    error: defaultLoadError,
+  } = useTemplateDefaultCpuMemory(vm);
+  const [updateInProcess, setUpdateInProcess] = React.useState<boolean>(false);
+  const [updateError, setUpdateError] = React.useState<string>();
+
+  const [memory, setMemory] = React.useState<number>();
+  const [cpuCores, setCpuCores] = React.useState<number>();
+  const [memoryUnit, setMemoryUnit] = React.useState<string>();
+  const [isDropdownOpen, setIsDropdownOpen] = React.useState<boolean>(false);
+
+  const resultVirtualMachine = React.useMemo(() => {
+    const updatedVM = produce<V1VirtualMachine>(vm, (vmDraft: V1VirtualMachine) => {
+      vmDraft.spec.template.spec.domain.resources.requests = {
+        ...vm?.spec?.template?.spec?.domain?.resources?.requests,
+        memory: `${memory}${memoryUnit}`,
+      };
+      vmDraft.spec.template.spec.domain.cpu.cores = cpuCores;
+    });
+    return updatedVM;
+  }, [vm, memory, cpuCores, memoryUnit]);
+
+  React.useEffect(() => {
+    if (vm?.metadata) {
+      const requests = vm?.spec?.template?.spec?.domain?.resources?.requests as {
+        [key: string]: string;
+      };
+      const { size, unit } = getMemorySize(requests?.memory);
+      setMemoryUnit(unit);
+      setMemory(size);
+      setCpuCores(getCPUCount(vm?.spec?.template?.spec?.domain?.cpu));
+    }
+  }, [vm]);
+
+  const onSubmit = () => {
```

Comment:
delete this

Reply:
error and loading states are being managed by the onSubmit, should I get those as props instead?